### PR TITLE
Feature/GitHub actions

### PR DESCRIPTION
**Pull request by balos1**

### CODE REVIEW

File: doc/sundials_developers/source/testing/Local.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   docker exec -w /sundials/test sundialsci-int32-double-latest ./test_driver.sh --testtype CUSTOM --env env/docker.sh --tpls --realtype double --indexsize 32
+
+Alternatively, you can drop into a bash shell inside the container to run specific examples:
+
+.. code-block:: shell
+
+   docker exec -it sundialsci-int32-double-latest bash
+
+On Macs, it is recommended to use `Podman <https://podman.io/>`_ (and then the
+same steps above apply. using ``podman`` instead of ``docker``). Podman is
```
Comment (drreynolds) [inline doc/sundials_developers/source/testing/Local.rst:45]: Extraneous period

File: docker/sundials-ci/int64-single/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        target: [x86_64]
+        blas:
+        - openblas+ilp64
+        mpi:
+        - openmpi
+      compiler:
+      - gcc@9.4.0
+  specs:
+  - cmake %gcc@9.4.0 arch=x86_64
+  # - petsc~double+int64 ^openblas+ilp64 %gcc@9.4.0 arch=x86_64
```
Comment (drreynolds) [inline docker/sundials-ci/int64-single/spack.yaml:14]: Should these lines be removed instead of commented out?

**Review by drreynolds — APPROVED** [review]
Aside from the incredibly minor comments I made, this looks excellent to me.

File: src/sundials/sundials_math.c [hunk 3]
```diff
@@ -106,3 +107,148 @@ booleantype SUNRCompareTol(realtype a, realtype b, realtype tol)
...
+  } else {
+    u.i--;
+  }
+
+  return u.f;
+#endif
+}
+
+
+long double SUNNextafterld(long double from, long double to)
```
[the remark this answers is not in the document]
Comment (balos1) [reply src/sundials/sundials_math.c:186]: No they are not, I added them for the sake of completeness, but I suppose we can drop the extra ones.

File: src/sundials/sundials_math.c [hunk 4]
```diff
@@ -106,3 +107,148 @@ booleantype SUNRCompareTol(realtype a, realtype b, realtype tol)
...
+
+sunrealtype SUNStrToReal(const char* str)
+{
+  char* end;
+#if (__STDC_VERSION__ >= 199901L)
+#if defined(SUNDIALS_EXTENDED_PRECISION)
+  /* Use strtod, but then round down to the closest double value
+     since strtod will effectively round up to the closest long double. */
+  double val = strtod(str, &end);
+  return (sunrealtype) SUNNextafterld(val, -0.0);
```
[the remark this answers is not in the document]
Comment (balos1) [reply src/sundials/sundials_math.c:231]: Should be using it here, I think it was a copy-paste error.

File: src/sundials/sundials_math.c [hunk 5]
```diff
@@ -106,3 +107,148 @@ booleantype SUNRCompareTol(realtype a, realtype b, realtype tol)
...
+  }
+
+  return u.f;
+#endif
+}
+
+sunrealtype SUNStrToReal(const char* str)
+{
+  char* end;
+#if (__STDC_VERSION__ >= 199901L)
```
[the remark this answers is not in the document]
Comment (balos1) [reply src/sundials/sundials_math.c:151]: Yes, we should. I will take care of this after the core problem is fixed with #156.

**Review by gardner48 — CHANGES_REQUESTED** [review]
One minor comment on the recent updates and there are a few unresolved minor comments from my earlier review.

File: doc/sundials_developers/source/testing/Local.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+   -----------------------------------------------------------------------------
+
+Testing Locally
+===============
+
+Using sundials-ci Docker Containers
+-----------------------------------
+
+It is possible to use the sundials-ci Docker containers available at
+`https://hub.docker.com/r/balos1/sundials-ci <https://hub.docker.com/r/balos1/sundials-ci>_`
```
[the remark this answers is not in the document]
Comment (balos1) [reply doc/sundials_developers/source/testing/Local.rst:21]: As per offline discussion, we will use the GitHub container registry instead.

File: test/test_driver.sh [hunk 7]
```diff
@@ -134,6 +134,18 @@ EOF
...
+# if [ ! -z "$GITHUB_WORKSPACE" ]; then
+#     # We need this to avoid git 'unsafe workspace' errors.
+#     git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+#     # GitHub CI workdir will be /github/workspace where the root
+#     # of SUNDIALS is workspace. We need to cd to the test/ dir.
+#     cd test/
+
+#     echo "change to $(pwd)"
+# fi
```
[the remark this answers is not in the document]
Comment (balos1) [reply test/test_driver.sh:147]: yep

File: doc/shared/versions.py [hunk 8]
```diff
@@ -16,3 +16,4 @@
 ida_version = 'v6.2.0'
 idas_version = 'v5.2.0'
 kinsol_version = 'v6.2.0'
+year = '2022'
```
Comment (gardner48) [inline doc/shared/versions.py:19]: New line at end of file

File: doc/superbuild/source/conf.py [hunk 9]
```diff
@@ -100,14 +100,14 @@ def setup(app: Sphinx):
 }
 
 rst_epilog = """
-.. |YEAR| replace:: 2021
+.. |YEAR| replace:: 2022
```
Comment (gardner48) [inline doc/superbuild/source/conf.py:103]:
```suggestion
.. |YEAR| replace:: {year}
```

File: docker/sundials-ci/build-all-containers.sh [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+images=('int32-double' 'int32-extended' 'int32-single' 'int64-double' 'int64-extended' 'int64-single')
+set -x
+docker build -t ghcr.io/llnl/sundials-ci-e4s-base:latest e4s-base/
+docker build -t ghcr.io/llnl/sundials-ci-e4s-base:e4s-22.05 e4s-base/
+for image in ${images[@]}; do
+  docker build -t ghcr.io/llnl/sundials-ci-${image}:latest --build-arg spack_yaml="${image}/spack.yaml" e4s-quarterly/
+done
+
+docker build -t ghcr.io/llnl/sundials-ci-${image}:spack-develop --build-arg spack_yaml="${image}/spack.yaml" spack-nightly/
+docker build -t ghcr.io/llnl/sundials-ci-${image}:spack-develop --build-arg spack_yaml="${image}/spack.yaml" spack-nightly/
```
Comment (gardner48) [inline docker/sundials-ci/build-all-containers.sh:14]: New line at end of file

**Review by gardner48 — CHANGES_REQUESTED** [review]
Super minor updates, otherwise LGTM

**Review by gardner48 — APPROVED** [review]

**Pull request merged**